src/model/MyIIMaze2DGenerator.java

```diff
@@ -0,0 +1,93 @@
+package model;
+
+import java.util.*;
+
+public class MyIIMaze2DGenerator extends AbstractIMaze2DGenerator {
+
+    @Override
+    public Maze2d generate(int mazeSize) {
+        Maze2d maze = new Maze2d(mazeSize);
+
+        // creating whole maze using random Prim's algorithm:
+        createRandMazePrim(maze);
+
+        return maze;
+    }
+
+    private void createRandMazePrim(Maze2d maze) {
+
+        Random rand = new Random();
+        MazePoint[][] pointsPath = new MazePoint[maze.mazeSize][maze.mazeSize];
+        ArrayList<MazePoint[]> mazeStructure = new ArrayList<>();
+
+        for (int i = 0; i < maze.mazeSize; i++)
+            for (int j = 0; j < maze.mazeSize; j++)
+                pointsPath[i][j] = new MazePoint(i, j);
+
+        // inserting start point to the maze structure:
+        MazePoint entranceMazePoint = pointsPath[maze.getEntrance().x][maze.getEntrance().y];
+        mazeStructure.add(new MazePoint[] {entranceMazePoint, entranceMazePoint});
+
+
+        // while there are unvisited available dual neighbors:
+        while (!mazeStructure.isEmpty()) {
+
+            // get random dual neighbor:
+            MazePoint[] currDualNeighbor = mazeStructure.remove(rand.nextInt(mazeStructure.size()));
+
+            // in case the SECOND nearest isn't visited (no path between them):
+            if(!pointsPath[currDualNeighbor[1].getX()][currDualNeighbor[1].getY()].isVisited()) {
+
+                // set both first nearest and second nearest as visited (create path):
+                pointsPath[currDualNeighbor[0].getX()][currDualNeighbor[0].getY()].setVisited(true);
+                pointsPath[currDualNeighbor[1].getX()][currDualNeighbor[1].getY()].setVisited(true);
+
+                // set visited both neighbors on myMaze:
+                maze.setPointVal(currDualNeighbor[0].getX(), currDualNeighbor[0].getY(), true);
+                maze.setPointVal(currDualNeighbor[1].getX(), currDualNeighbor[1].getY(), true);
+
+                // get all dual available neighbors:
+                getDualAvailNeighbors(maze, pointsPath, currDualNeighbor, mazeStructure);
+
+            }
+        }
+        // set the exit point as a valid path:
+        maze.setPointVal((int) maze.exit.getX(), (int) maze.exit.getX(), true);
+
+        // While using even maze, we should clear manually the cell near the exit:
```

Comment:
Ok just be consistent with first big letter/not

Reply:
Sure, I'll change to a capital letter in all comments of the project